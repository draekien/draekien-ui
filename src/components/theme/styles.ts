export const styles = {
  root: {
    fontFamily: 'Montserrat, sans-serif',
    variant: 'text.medium',
    '*:focus': {
      outline: 'none',
    },
  },
};
