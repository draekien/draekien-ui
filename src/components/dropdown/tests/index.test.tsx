/** @jsxRuntime classic */
/** @jsx jsx */
import { jsx } from 'theme-ui';
import { render } from '@testing-library/react';
import Dropdown from '..';
import { caretCss } from '../index.styles';

describe('Dropdown component', () => {
  test('should match snapshot', () => {
    const { container } = render(
      <Dropdown label="dropdown">
        {[1, 2, 3, 4, 5, 6, 7].map((x) => (
          <div key={x}>Item {x}</div>
        ))}
      </Dropdown>
    );

    expect(container).toMatchSnapshot();
  });

  test('should match snapshot when hasSelection is true and position is top right', () => {
    const { container } = render(
      <Dropdown label="dropdown" position="top-right" hasSelection>
        {[1, 2, 3, 4, 5, 6, 7].map((x) => (
          <div key={x}>Item {x}</div>
        ))}
      </Dropdown>
    );

    expect(container).toMatchSnapshot();
  });

  test('should match snapshot when variant is primary and position is bottom right', () => {
    const { container } = render(
      <Dropdown label="dropdown" position="bottom-right" variant="primary">
        {[1, 2, 3, 4, 5, 6, 7].map((x) => (
          <div key={x}>Item {x}</div>
        ))}
      </Dropdown>
    );

    expect(container).toMatchSnapshot();
  });

  test('should match snapshot when maxHeight is set', () => {
    const { container } = render(
      <Dropdown label="dropdown" maxHeight="20px">
        {[1, 2, 3, 4, 5, 6, 7].map((x) => (
          <div key={x}>Item {x}</div>
        ))}
      </Dropdown>
    );

    expect(container).toMatchSnapshot();
  });

  test('should match snapshot when isOpen is true', () => {
    const { container } = render(
      <Dropdown label="dropdown" isOpen>
        {[1, 2, 3, 4, 5, 6, 7].map((x) => (
          <div key={x}>Item {x}</div>
        ))}
      </Dropdown>
    );

    expect(container).toMatchSnapshot();
  });

  test('caretCss should match snapshot', () => {
    expect(caretCss({ isOpen: false, margin: 4 })).toMatchSnapshot();
  });
});
