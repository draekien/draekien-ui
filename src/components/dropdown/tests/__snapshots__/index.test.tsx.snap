// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component caretCss should match snapshot 1`] = `
Object {
  "borderLeft": "4px solid transparent",
  "borderRight": "4px solid transparent",
  "borderTop": "4px solid",
  "display": "inline-block",
  "height": 0,
  "marginLeft": "0.5rem",
  "verticalAlign": "middle",
  "width": 0,
}
`;

exports[`Dropdown component should match snapshot 1`] = `
<div>
  <div
    class="css-l3v2wu-Dropdown"
  >
    <div
      style="display: inline-block;"
    >
      <div
        class="css-s2scu3-Dropdown"
      >
        <button
          class="css-1eaah62-Button"
          type="button"
        >
          <span>
            dropdown
          </span>
          <span
            class="css-bq4wcj"
          >
            <i
              class="material-icons  css-18og482-Icon"
            >
              expand_more
            </i>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dropdown component should match snapshot when hasSelection is true and position is top right 1`] = `
<div>
  <div
    class="css-l3v2wu-Dropdown"
  >
    <div
      style="display: inline-block;"
    >
      <div
        class="css-165ucqk-Dropdown"
      >
        <button
          class="css-1eaah62-Button"
          type="button"
        >
          <span>
            dropdown
          </span>
          <span
            class="css-bq4wcj"
          >
            <i
              class="material-icons  css-18og482-Icon"
            >
              expand_more
            </i>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dropdown component should match snapshot when isOpen is true 1`] = `
<div>
  <div
    class="css-l3v2wu-Dropdown"
  >
    <div
      style="display: inline-block;"
    >
      <div
        class="css-s2scu3-Dropdown"
      >
        <button
          class="css-1iozysr-Button"
          type="button"
        >
          <span>
            dropdown
          </span>
          <span
            class="css-bq4wcj"
          >
            <i
              class="material-icons  css-18og482-Icon"
            >
              expand_less
            </i>
          </span>
        </button>
        <div
          class="css-1ker5np-Dropdown"
        >
          <div>
            Item 
            1
          </div>
          <div>
            Item 
            2
          </div>
          <div>
            Item 
            3
          </div>
          <div>
            Item 
            4
          </div>
          <div>
            Item 
            5
          </div>
          <div>
            Item 
            6
          </div>
          <div>
            Item 
            7
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dropdown component should match snapshot when maxHeight is set 1`] = `
<div>
  <div
    class="css-l3v2wu-Dropdown"
  >
    <div
      style="display: inline-block;"
    >
      <div
        class="css-s2scu3-Dropdown"
      >
        <button
          class="css-1eaah62-Button"
          type="button"
        >
          <span>
            dropdown
          </span>
          <span
            class="css-bq4wcj"
          >
            <i
              class="material-icons  css-18og482-Icon"
            >
              expand_more
            </i>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dropdown component should match snapshot when variant is primary and position is bottom right 1`] = `
<div>
  <div
    class="css-l3v2wu-Dropdown"
  >
    <div
      style="display: inline-block;"
    >
      <div
        class="css-s2scu3-Dropdown"
      >
        <button
          class="css-2531nh-Button"
          type="button"
        >
          <span>
            dropdown
          </span>
          <span
            class="css-bq4wcj"
          >
            <i
              class="material-icons  css-18og482-Icon"
            >
              expand_more
            </i>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
